### PR TITLE
New endpoints for moderated requests flows

### PR DESCRIPTION
**Pull request by m1lt0n**
#### Summary

This PR is about introducing a set of new flows and endpoints for storing requests for action (e.g. deleting a workspace, etc) in pending state, allowing moderation/approval of such requests instead of executing them right away.

In the process, what has been added by this PR:

1. Gorm ORM for storing the moderated requests in PostgreSQL
2. Goose database migration tool for changing the schema of the database
3. docker-compose.yaml with a PostgreSQL database service for local development
4. A validation library for validating user input in the moderated request creation endpoint

**Endpoints created**
1. POST /moderated-requests : create a request for some action (currently WORKSPACES_DELETE)
2. GET /moderated-requests : list all moderated requests

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: moderated_requests/dto.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+const (
+	Pending  State = "PENDING"
+	Rejected State = "REJECTED"
+	Approved State = "APPROVED"
+)
+
+type Payload = map[string]interface{}
+
+type ModeratedRequestData struct {
+	Kind Kind    `json:"kind" validate:"oneof=WORKSPACES_DELETE"`
```
Comment (spirosoik) [inline moderated_requests/dto.go]: one easy way yo avoid the strings here for the state, is to set a binding custom validator.

File: moderated_requests/service.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		logger: logger,
+		db:     db,
+	}
+}
+
+func (s *Service) CreateRequest(req ModeratedRequestData) error {
+	data, err := json.Marshal(req.Data)
+
+	if err != nil {
+		return err
```
Comment (spirosoik) [inline moderated_requests/service.go]: it is a good practice to wrap the error `errors.Wrap()`. Details here https://dave.cheney.net/practical-go/presentations/qcon-china.html#_wrapping_errors_with_github_compkgerrors

**Review by spirosoik — COMMENTED** [review]
Good job 🎉 . Few minor comments

File: moderated_requests/dto.go (hunk 1 above)
Comment (m1lt0n) [reply moderated_requests/dto.go]: yep :) thanks for both comments, I've made changes on both errors and used a custom validator.

**Review by spirosoik — APPROVED** [review]
LGTM

**Review by spirosoik — APPROVED** [review]

**Pull request merged**